airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py

```diff
@@ -1034,6 +1034,8 @@ def add_to_outputs(
             if suffix == "scd":
                 stg_schema = self.get_schema(True)
                 stg_table = self.tables_registry.get_file_name(schema, self.json_path, self.stream_name, "stg", truncate_name)
+                if stg_table[0].isdigit():
+                    stg_table = f'"{stg_table}"'
```

Comment:
Why is the fix only on "stg" tables not on standard table names too? 

If I understand, you are forcing quoting into the SQL model file name... I'm not sure why do you take this approach though?
Does this apply regardless of the destination type?

Shouldn't this be configured through https://docs.getdbt.com/reference/project-configs/quoting maybe?

the get_file_name method is making a final call to `normalize_table_name`, shouldn't this digit check happen there instead?